lib/src/test/java/io/ably/lib/test/realtime/ConnectionManagerTest.java

```diff
@@ -408,9 +408,10 @@ public void onConnectionStateChanged(ConnectionStateChange state) {
 					} catch (NoSuchFieldException|IllegalAccessException e) {
 						fail("Unexpected exception in checking connectionStateTtl");
 					}
-					ably.close();
 				}
 			});
+			new Helpers.ConnectionManagerWaiter(ably.connection.connectionManager).waitFor(ConnectionState.connected);
```

Comment:
I agree this is an improvement, but even better would have been to have an explicit check that the callback ran

Reply:
@paddybyers done https://github.com/ably/ably-java/pull/389/commits/9de807f16a0dc59c228b2701db163f23f752dfc8